### PR TITLE
Add Standalone Mixture of Experts example

### PR DESCRIPTION
**Pull request by Abhinavexists**
Add standalone Mixture-of-Experts (MoE) layers for Keras, usable as drop-in replacements for Dense and Conv2D. Includes full example on CIFAR-10 demonstrating:

 - DenseMoE: soft-routed expert networks for fully connected layers
 - Conv2DMoE: convolutional expert networks with 1x1 gating
 - Expert specialization through softmax gating
 - Performance comparison against a standard baseline model
 - Expert utilization inspection via gating statistics

### CODE REVIEW

File: examples/keras_recipes/mixture_of_experts.py [hunk 1]
```diff
@@ -0,0 +1,629 @@
...
+    def __init__(
+        self,
+        units,
+        n_experts,
+        expert_activation=None,
+        gating_activation=None,
+        use_expert_bias=True,
+        use_gating_bias=True,
+        **kwargs,
+    ):
```
[the remark this answers is not in the document]
Comment (Abhinavexists) [reply examples/keras_recipes/mixture_of_experts.py:82]: i want to keep it none so it provides flexibilty to user in what activation they want to use

File: examples/keras_recipes/mixture_of_experts.py [hunk 2]
```diff
@@ -0,0 +1,629 @@
...
+        kernel_size,
+        n_experts,
+        strides=1,
+        padding="valid",
+        expert_activation=None,
+        gating_activation=None,
+        use_expert_bias=True,
+        use_gating_bias=True,
+        **kwargs,
+    ):
```
[the remark this answers is not in the document]
Comment (Abhinavexists) [reply examples/keras_recipes/mixture_of_experts.py:207]: i want to keep it none so it provides flexibilty to user in what activation they want to use

File: examples/keras_recipes/mixture_of_experts.py [hunk 3]
```diff
@@ -0,0 +1,629 @@
...
+inputs across experts. Let's examine the gating weights to understand expert
+utilization patterns.
+"""
+
+# Get the first DenseMoE layer from the MoE model
+dense_moe_layer = None
+for layer in moe_model.layers:
+    if isinstance(layer, DenseMoE):
+        dense_moe_layer = layer
+        break
```
[the remark this answers is not in the document]
Comment (Abhinavexists) [reply examples/keras_recipes/mixture_of_experts.py:546]: no need as it takes away the simplicity of implementation

File: examples/keras_recipes/mixture_of_experts.py [hunk 4]
```diff
@@ -0,0 +1,629 @@
...
+    )
+
+    # Get features for a sample of test data
+    test_features = feature_extractor.predict(x_test[:100], verbose=0)
+
+    # Compute gating weights directly using the layer's learned parameters
+    gating_logits = ops.tensordot(test_features, dense_moe_layer.gating_kernel, axes=1)
+    if dense_moe_layer.use_gating_bias:
+        gating_logits = gating_logits + dense_moe_layer.gating_bias
+    gating_weights = dense_moe_layer.gating_activation(gating_logits)
```
[the remark this answers is not in the document]
Comment (Abhinavexists) [reply examples/keras_recipes/mixture_of_experts.py:559]: lets keep direct compution for this example , i think it works

File: examples/keras_recipes/mixture_of_experts.py [hunk 5]
```diff
@@ -0,0 +1,629 @@
...
+        expert_outputs = self.expert_activation(expert_outputs)
+
+        # Compute gating weights
+        # Shape: (batch_size, n_experts)
+        gating_outputs = ops.tensordot(inputs, self.gating_kernel, axes=1)
+
+        if self.use_gating_bias:
+            gating_outputs = gating_outputs + self.gating_bias
+
+        gating_outputs = self.gating_activation(gating_outputs)
```
[the remark this answers is not in the document]
Comment (Abhinavexists) [reply examples/keras_recipes/mixture_of_experts.py:137]: No need

**Comment by Abhinavexists** [comment]
Hi @sachinprasadhs 
is there any changes required for this pr as its still unmerged ?
